### ISSUE TITLE
Unable to validate Indian phone numbers

### ISSUE THREAD

**Issue by harikrishnanng** [Propaganistas/Laravel-Phone#81]
Unable to validate some Indian mobile numbers starting with 81,83,85 etc. Gives the error:
`{
    "message": "The given data was invalid.",
    "errors": {
        "phone_no": [
            "The phone number field contains an invalid number."
        ]
    }
}`

//validator in RegisterController.php

<pre>
<code>
protected function validator(array $data)
    {
        return Validator::make($data, [
            'name' => 'required|string|max:255',
            'email' => 'required|string|email|max:255|unique:users',
            'user_name' => 'required|string|min:6|unique:users',
            'gender' => 'required|string|in:male,female,other',
            'password' => 'required|string|min:6|confirmed',
            'cc' => 'required', 
        ])->setAttributeNames([
            'phone_no' => 'phone number',
            'cc' => 'country code'
        ]);
    }
</code>
</pre>

>cc is the country code passed from UI (IN,US etc), phone_no is just 10 digits

//registration func
<pre>
<code>
public function register(Request $request)
{
    $this->validator($request->all())->sometimes('phone_no', 'required|phone:'.mb_strtoupper($request->cc).',mobile|unique:users', function ($input){
        return $input->cc;
    })->validate();

    event(new Registered($user = $this->create($request->all())));

    $this->guard()->login($user);

    if($request->isJson() && !empty(Auth::user())) {
        return response()->json(['code' => 200, 'token' => Auth::user()->api_token,'role' => Auth::user()->getRoleNames(), 'message' => 'success']);
    }

    return $this->registered($request, $user)
    ?: redirect($this->redirectPath());
}
</code>
</pre>

//numbers used
81/83/8590332334 
same number starting with 82,84,90,70 are accepted

**Comment by Propaganistas** [comment]
If you expect to receive support you'll want to include some example numbers and sample code.

**Comment by harikrishnanng** [comment]
I'm using the latest version

**Comment by Propaganistas** [comment]
The number you supplied validates just fine using `IN` as country. So there's probably something wrong with your code. Dump the value of `$request->cc` to check if it really holds `IN`.

Additionally, you might want to use the built-in functionality to using the country field instead of directly inlining `$request->cc` in the validation rule. Just reference the country field's name.

```php
$this->validator($request->all())->sometimes('phone_no', 'required|phone:cc,mobile|unique:users', function ($input){
    return $input->cc;
})->validate();
```

**Comment by harikrishnanng** [comment]
![capture](https://user-images.githubusercontent.com/19324484/37135664-4d99acdc-22c4-11e8-90cd-7c53eeb1a07c.PNG)

<pre>
<code>
Validator::make($request->all(), [
            'phone_no' => 'required|phone:cc,mobile|unique:users'
        ])->setAttributeNames(['phone_no' => 'phone number'])->validate();
</code>
</pre>

**Comment by Propaganistas** [comment]
Bug found and fixed in 79b7901. New release created. Thanks for reporting.
